Fatal: interface {} is nil, not *rest.Config

# What's happening

The config operator is in CrashLoopBack state.

Seems to happen here : https://github.com/redhat-cop/vault-config-operator/blob/main/api/v1alpha1/utils/commons.go#L91

It used to work. I think it's only happening since recent versions  (https://github.com/redhat-cop/vault-config-operator/pull/72/files#diff-ffe0693678bc581dd545abd5b60a728b97bf196ca911806f8490383b73ebfd35R91)

```
Observed a panic in reconciler: interface conversion: interface {} is nil, not *rest.Config	{"controller": "databasesecretengineconfig", "controllerGroup": "[redhatcop.redhat.io](http://redhatcop.redhat.io/)", "controllerKind": "DatabaseSecretEngineConfig", "databaseSecretEngineConfig": {"name":"mariadb","namespace":"xxx-dev"}, "namespace": "xxx-dev", "name": "mariadb", "reconcileID": "07e6212a-ffe3-4d6f-a941-ad2edab89e18"}
panic: interface conversion: interface {} is nil, not *rest.Config [recovered]
	panic: interface conversion: interface {} is nil, not *rest.Config
goroutine 1320 [running]:
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1())
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x1840e00, 0xc008890900})
	/opt/hostedtoolcache/go/1.18.5/x64/src/runtime/panic.go:838 +0x207
[github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.GetJWTTokenWithDuration({0x1d07710](http://github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.GetJWTTokenWithDuration(%7B0x1d07710), 0xc008890780}, {0xc00067b237, 0x7}, {0xc00067b240, 0xe}, 0x258)
	/home/runner/work/vault-config-operator/vault-config-operator/api/v1alpha1/utils/commons.go:91 +0x3d0
[github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.GetJWTToken(...)](http://github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.GetJWTToken(...))
	/home/runner/work/vault-config-operator/vault-config-operator/api/v1alpha1/utils/commons.go:118
[github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.(*KubeAuthConfiguration).getJWTToken(0x1d07710](http://github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.(*KubeAuthConfiguration).getJWTToken(0x1d07710)?, {0x1d07710?, 0xc008890780?}, {0xc00067b240?, 0x0?})
	/home/runner/work/vault-config-operator/vault-config-operator/api/v1alpha1/utils/commons.go:122 +0x4a
[github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.(*KubeAuthConfiguration).GetVaultClient(0xc0000e8618](http://github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils.(*KubeAuthConfiguration).GetVaultClient(0xc0000e8618), {0x1d07710, 0xc008890780}, {0xc00067b240, 0xe})
	/home/runner/work/vault-config-operator/vault-config-operator/api/v1alpha1/utils/commons.go:75 +0xa7
[github.com/redhat-cop/vault-config-operator/controllers.(*DatabaseSecretEngineConfigReconciler).Reconcile(0xc0000cc1c0](http://github.com/redhat-cop/vault-config-operator/controllers.(*DatabaseSecretEngineConfigReconciler).Reconcile(0xc0000cc1c0), {0x1d07710, 0xc008890780}, {{{0xc00067b240?, 0x10?}, {0xc00067b230?, 0x40f367?}}})
	/home/runner/work/vault-config-operator/vault-config-operator/controllers/databasesecretengineconfig_controller.go:84 +0x165
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1d07668](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1d07668)?, {0x1d07710?, 0xc008890780?}, {{{0xc00067b240?, 0x1977600?}, {0xc00067b230?, 0x404c94?}}})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:121 +0xc8
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00037e460](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00037e460), {0x1d07668, 0xc0002cc140}, {0x18928a0?, 0xc000832a80?})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:320 +0x33c
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00037e460](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00037e460), {0x1d07668, 0xc0002cc140})
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:273 +0x1d9
[sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2())
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:234 +0x85
created by [sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2](http://sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.12.2/pkg/internal/controller/controller.go:230 +0x325
```

# Environment

* Vault config operator 0.6.4
* Openshift 4.9.37
* Vault 1.11.3


Thanks,

--- a/controllers/databasesecretengineconfig_controller.go
+++ b/controllers/databasesecretengineconfig_controller.go
@@ -80,13 +80,12 @@ func (r *DatabaseSecretEngineConfigReconciler) Reconcile(ctx context.Context, re
 		return reconcile.Result{}, err
 	}
 
-	ctx1 := context.WithValue(ctx, "kubeClient", r.GetClient())
-	vaultClient, err := instance.Spec.Authentication.GetVaultClient(ctx, instance.Namespace)
+	ctx1, err := prepareContext(ctx, r.ReconcilerBase, instance)
 	if err != nil {
-		r.Log.Error(err, "unable to create vault client", "instance", instance)
+		r.Log.Error(err, "unable to prepare context", "instance", instance)
 		return r.ManageError(ctx, instance, err)
 	}
-	ctx1 = context.WithValue(ctx1, "vaultClient", vaultClient)
+
 	vaultResource := vaultresourcecontroller.NewVaultResource(&r.ReconcilerBase, instance)
 
 	return vaultResource.Reconcile(ctx1, instance)

